internal/searcher/client.go

```diff
@@ -32,31 +32,40 @@ var cachedClient *client
 func Client() *client {
 	if cachedClient == nil {
 		endpoint, _ := url.JoinPath(searchHost(), "search")
+		// TODO: how to handle error
 		cachedClient = &client{
 			endpoint: endpoint,
 		}
 	}
 	return cachedClient
 }
 
-func (c *client) Search(query string) (*SearchResult, error) {
-	return execSearch(c.endpoint + "?q=" + url.QueryEscape(query))
+func (c *client) Search(query string, options ...func() string) (*SearchResult, error) {
+	if query == "" {
+		return nil, fmt.Errorf("query should not be empty")
+	}
+
+	searchURL := c.endpoint + "?q=" + url.QueryEscape(query)
+
+	for _, op := range options {
+		searchURL += op()
+	}
+
+	return execSearch(searchURL)
 }
 
-func (c *client) SearchVersion(query, version string) (*SearchResult, error) {
-	return execSearch(
-		c.endpoint +
-			"?q=" + url.QueryEscape(query) +
-			"&v=" + url.QueryEscape(version),
-	)
+func WithVersion(version string) func() string {
```

Comment:
I think defining a `type SearchOption func() string` would help here.

As-is, `searcher.WithVersion` is a ambiguous function since its not clear what context it needs to be used in. By adding this type definition, readers can see the return type `SearchOption` and then look for functions that accept `SearchOption` parameters.

Reply:
Exactly.

Reply:
Now that this PR is merged, I'll create a new PR to update it.

Reply:
https://github.com/jetpack-io/devbox/pull/1173